webapp/src/modules/mortgage/reducer.js

```diff
@@ -0,0 +1,66 @@
+import { combineReducers } from 'redux'
+import { loadingReducer } from 'modules/loading/reducer'
+
+import {
+  FETCH_MORTGAGED_PARCELS_REQUEST,
+  FETCH_MORTGAGED_PARCELS_SUCCESS,
+  FETCH_MORTGAGED_PARCELS_FAILURE,
+  FETCH_ACTIVE_PARCEL_MORTGAGES_REQUEST,
+  FETCH_ACTIVE_PARCEL_MORTGAGES_FAILURE,
+  FETCH_ACTIVE_PARCEL_MORTGAGES_SUCCESS
+} from './actions'
+
+const INITIAL_DATA_STATE = {
+  mortgages: {}
+}
+
+export function data(state = INITIAL_DATA_STATE, action) {
+  switch (action.type) {
+    case FETCH_MORTGAGED_PARCELS_SUCCESS:
+    case FETCH_ACTIVE_PARCEL_MORTGAGES_SUCCESS: {
+      return {
+        ...state,
+        mortgages: action.mortgages.reduce(
+          (normalizedMortgages, mortgage) => ({
+            ...normalizedMortgages,
+            [mortgage.tx_hash]: mortgage
+          }),
+          state.mortgages
+        )
+      }
+    }
+    default:
+      return state
+  }
+}
+
+export function loading(state = [], action) {
+  switch (action.type) {
+    case FETCH_MORTGAGED_PARCELS_SUCCESS:
+    case FETCH_ACTIVE_PARCEL_MORTGAGES_SUCCESS:
+    case FETCH_ACTIVE_PARCEL_MORTGAGES_FAILURE:
+    case FETCH_MORTGAGED_PARCELS_FAILURE:
+    case FETCH_MORTGAGED_PARCELS_REQUEST:
+    case FETCH_ACTIVE_PARCEL_MORTGAGES_REQUEST: {
+      return loadingReducer(state, action)
+    }
+    default:
+      return state
+  }
+}
+
+export function error(state = false, action) {
+  switch (action.type) {
+    case FETCH_ACTIVE_PARCEL_MORTGAGES_FAILURE:
+    case FETCH_MORTGAGED_PARCELS_FAILURE:
+      return action.error
+    default:
+      return state
+  }
+}
+
+export default combineReducers({
+  data,
+  loading,
+  error
+})
```

Comment:
I like this approach! buuut (sorry 😭) I'm not sure about having it just here. Every other reducer uses only one function.
@cazala I'll leave the veredict to you.

Reply:
I told @cazala about it. He was ok

Reply:
Yeap me too!, my problem was opening this file and encountering a different implementation from all the other reducers.
I don't want to be rigid tho, but sometimes I prefer keeping an slighty more cumbersome implementation if that means easier induction of new people to the project and removing the possibility of having yet another implementation later.

That said, 👍

Reply:
Thanks, I would like To propose this “new” way to architecting reducers if we all agree obviusly